### PR TITLE
Replace ListConstruct on inputs to Roll and Broadcast

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Resolve pattern when `prim::ListConstruct` is used as list of scalars which can be converted to `Concat`*

### Details:
 - *It is very hard to create a simple model with list of scalars to test this in layer tests. This was tested locally on `swin_b` model from `torchvision` and `vgg11` from `timm`*

### Tickets:
 - *None*

### CODE REVIEW

File: src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using namespace ov::pass;
+using namespace ov::op;
+
+ListConstructReplacer::ListConstructReplacer() {
+    // Transformation for torch operators for cases where prim::ListConstruct can be replaced with Concat.
+    auto list_construct = pattern::wrap_type<ov::op::util::FrameworkNode>();
+
+    // Both aten::view and aten::reshape are using same translation returning Reshape operator.
+    auto reshape_op = pattern::wrap_type<v1::Reshape>({pattern::any_input(), list_construct});
+    //auto roll_op = pattern::wrap_type<v7::Roll>({pattern::any_input(), list_construct, pattern::any_input()});
```
Comment (eaidova) [inline src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp:34]: does it some debug comments and should be removed?

Comment (mvafin) [reply src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp:34]: Good catch, yeah, that is artifacts from testing

Comment (mvafin) [inline src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp:34]:
```suggestion
    auto roll_op = pattern::wrap_type<v7::Roll>({pattern::any_input(), list_construct, pattern::any_input()});
```

File: src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    auto broadcast_op = pattern::wrap_type<v1::Broadcast>({pattern::any_input(), list_construct});
+
+    auto lc_pattern = std::make_shared<pattern::op::Or>(OutputVector{reshape_op, /*roll_op,*/ broadcast_op});
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/listconstruct_replacer.cpp:37]:
```suggestion
    auto lc_pattern = std::make_shared<pattern::op::Or>(OutputVector{reshape_op, roll_op, broadcast_op});
```

**Review by eaidova — APPROVED** [review]

**Pull request merged**